src/cxristian/Matriz_Cuadrada.java

```diff
@@ -0,0 +1,100 @@
+import java.util.Scanner;
+
+public class Matriz_Cuadrada {
+    private Scanner keyboard;
+    private int[][] mat;
+    int[] diagoPrincipal;
+    int[] diagoSecundaria;
+    int suma1 = 0, suma2 = 0;
+
+    public static void main(String[] ar) {
+        Matriz_Cuadrada ma=new Matriz_Cuadrada();
+        //ma.uploadNumbers();
+        ma.uploadNumbersRandom();
+        ma.analyzeDiagonals();
+        System.out.println("\n" + ma.winer());
+    }
+
+    public void uploadNumbers() {
+        keyboard=new Scanner(System.in);
+        System.out.print("Cuantas fila y columnas tendra la matriz la matriz:");
+        int rows = keyboard.nextInt();
+        int columns = rows;
+        mat=new int[rows][columns];
+        diagoPrincipal = new int[mat.length];
+        diagoSecundaria = new int[mat.length];
+        for(int f=0;f<mat.length;f++) {
+            for(int c=0;c<mat[f].length;c++) {
+                System.out.print("Ingrese componente: ");
+                mat[f][c]=keyboard.nextInt();
+            }
+        }
+        printMatriz(mat);
+    }
+
+    public void uploadNumbersRandom() {
+        keyboard=new Scanner(System.in);
+        System.out.print("Cuantas fila y columnas tendra la matriz la matriz:");
+        int rows = keyboard.nextInt();
+        int columns = rows;
+        mat=new int[rows][columns];
+        diagoPrincipal = new int[mat.length];
+        diagoSecundaria = new int[mat.length];
+        for(int f=0;f<mat.length;f++) {
+            for(int c=0;c<mat[f].length;c++) {
```

Comment:
Considering the way this matrix is built (always square), maybe you don't need to ask for length, just use `rows` or `columns`